test/unit/profile_test.rb

```diff
@@ -12,139 +13,102 @@ class ProfileTest < ActiveSupport::TestCase
-    end
-
-    context '#company_type validation' do
-
-      context 'a profile of an individual' do
-        setup { @individual_profile.company_type = nil }
-
-        should 'be valid without company_type' do
-          assert @individual_profile.valid?
-        end
-      end
-
-      context 'a profile of a company' do
-        setup { @company_profile.company_type = nil }
-
-        should 'not be valid without company_type' do
-          assert_equal false, @company_profile.valid?
-          assert_equal false, @company_profile.errors[:company_type].empty?
-        end
-      end
+  test 'a profile of an individual should be valid without company_type' do
+    individual_profile = FactoryBot.create(:profile, company_size: Profile::IndividualNotCompany, company_type: nil)
+    assert_valid individual_profile
+  end
 
-    end
+  test 'a profile of a company should not be valid without company_type' do
+    company_profile = FactoryBot.build(:profile, company_size: Profile::CompanySizes.last, company_type: nil)
+    assert_not company_profile.valid?
+    assert_not_empty company_profile.errors[:company_type]
+  end
 
-    context '#individual_profile? method' do
-      should 'return true for individual profiles' do
-        assert @individual_profile.individual_profile?
-      end
+  test '#individual_profile? method should return true for individual profiles' do
+    individual_profile = FactoryBot.create(:profile, company_size: Profile::IndividualNotCompany)
+    assert individual_profile.individual_profile?
+  end
 
-      should 'return false for company profiles' do
-        assert_equal false, @company_profile.individual_profile?
-      end
+  test '#individual_profile? method should return false for company profiles' do
+    company_profile = FactoryBot.build(:profile, company_size: Profile::CompanySizes.last)
+    assert_not company_profile.individual_profile?
+  end
 
-      should 'return false for profiles with company_size set to nil' do
-        profile = Profile.new
-        profile.company_size = nil
+  test '#individual_profile? method should return false for profiles with company_size set to nil' do
+    # TODO: validation says company_size can't be blank, so is this test necessary/valid?
+    profile = Profile.new
+    profile.company_size = nil
 
-        assert_equal false, profile.individual_profile?
-      end
+    assert_not profile.individual_profile?
+    # assert_not Profile.new(company_size: nil).individual_profile?
+  end
 
-      should 'return false for profiles with company_size set to blank' do
-        assert_equal false, Profile.new(:company_size => "").individual_profile?
-      end
-    end
+  test '#individual_profile? method should return false for profiles with company_size set to blank' do
+    # TODO: validation says company_size can't be blank, so is this test necessary/valid?
+    assert_not Profile.new(company_size: '').individual_profile?
```

Comment:
it is possible that we have outdated data in DB so model might be loaded from DB with such data... hypothetically :)

But yes - it makes little sense.